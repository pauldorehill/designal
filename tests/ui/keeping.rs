use designal::Designal;

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean {
    #[forbid(dead_code)]
    taste: String,
}

#[derive(Designal)]
#[designal(trim_start = "Human")]
#[forbid(dead_code)]
struct HumanBean1 {
    crunch: String,
}

fn main() {
    let _ = Bean1 {
        crunch: String::new(),
    };
}
