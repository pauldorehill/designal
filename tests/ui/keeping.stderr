error: field is never read: `taste`
 --> $DIR/keeping.rs:7:5
  |
7 |     taste: String,
  |     ^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> $DIR/keeping.rs:6:14
  |
6 |     #[forbid(dead_code)]
  |              ^^^^^^^^^

error: field is never read: `taste`
 --> $DIR/keeping.rs:7:5
  |
7 |     taste: String,
  |     ^^^^^^^^^^^^^

error: struct is never constructed: `HumanBean1`
  --> $DIR/keeping.rs:13:8
   |
13 | struct HumanBean1 {
   |        ^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/keeping.rs:12:10
   |
12 | #[forbid(dead_code)]
   |          ^^^^^^^^^

error: field is never read: `crunch`
  --> $DIR/keeping.rs:14:5
   |
14 |     crunch: String,
   |     ^^^^^^^^^^^^^^
